### PR TITLE
The whole thing

### PR DESCRIPTION
**Pull request by causal-agent**
All of it.

### CODE REVIEW

File: app/authorization.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+var restify = require('restify');
+
+module.exports = function(apiKeys) {
+  return function(req, res, next) {
+    var authorized =
+      req.authorization &&
+      req.authorization.basic &&
+      apiKeys.hasOwnProperty(req.authorization.basic.username) &&
```
Comment (kiasaki) [inline app/authorization.js:10]: should you be lowercasing the username as that's how it's stored in the apiKeys hash?

File: app/config.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  var apiKeyVars = R.filter(R.compose(R.eq(0), R.strIndexOf('API_KEY_')), R.keys(env));
+  if (apiKeyVars.length) {
+    obj.apiKeys = {};
+    R.forEach(function(envVar) {
+      var client = R.toLower(R.substringFrom(8, envVar));
+      obj.apiKeys[client] = env[envVar];
+    }, apiKeyVars);
+  }
+
+  exports.loadObject(obj);
```
Comment (kiasaki) [inline app/config.js:64]: I like the pattern of that module being the config and not passing around an object in all calls that need the loaded config. (at the same time it's like using a global. but, hey, in an app that small, no harm done

File: app/controller.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+var R = require('ramda');
+var Promise = require('bluebird');
+var restify = require('restify');
+
+var cipher = require('./cipher');
+var vault = require('./vault');
+var config = require('./config');
```
Comment (kiasaki) [inline app/controller.js:9]: We should stop aligning our equals in napi...

File: app/authorization.js (hunk 1 above)
Comment (causal-agent) [inline app/authorization.js:10]: It’s lowercased in `apiKeys` because clients are expected to send it lowercase.

File: app/config.js (hunk 2 above)
Comment (causal-agent) [inline app/config.js:64]: Right, I actually only access `config` directly in the controller and the rotation script. Every other piece is re-usable without this module.

File: app/index.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+if (process.env.NODE_ENV === 'production') {
+  app.use(authorization(config.apiKeys));
+}
+
+app.post('/vault', controller.post);
+app.get('/vault/:id', controller.get);
+app.put('/vault/:id', controller.put);
+app.del('/vault/:id', controller.delete);
+
+if (require.main === module) {
```
Comment (kiasaki) [inline app/index.js:43]: I have always seen `!module.parent` used, how is this statement different?

Comment (causal-agent) [inline app/index.js:43]: It’s what the documentation recommends: https://nodejs.org/api/modules.html#modules_accessing_the_main_module https://nodejs.org/api/modules.html#modules_accessing_the_main_module

File: package.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "sesame",
+  "version": "0.0.0",
```
Comment (kiasaki) [inline package.json:3]: Let's version bump when it's push in prod and in a usable by napi state, if it's already there let's give it a version now

Comment (causal-agent) [inline package.json:3]: It should be 0.1.0 right now since it’s deployed in staging.

**Pull request closed without merging**